components/odh-notebook-controller/controllers/notebook_webhook.go

```diff
@@ -730,18 +730,28 @@ func SetContainerImageFromRegistry(ctx context.Context, config *rest.Config, not
 					}
 
 					imagestreamFound := false
-					// List imagestreams in the specified namespace
+					// Check for controller namespace before the notebook namespace
+					// to avoid user overriding the image selection in the case of
+					// images having the same name in both namespaces.
+
+					// List imagestreams in the controller namespace
 					imagestreams, err := dynamicClient.Resource(ims).Namespace(namespace).List(ctx, metav1.ListOptions{})
-					if err != nil {
-						if k8serr.IsForbidden(err) {
-							log.Info("Permission denied to list imagestreams", "namespace", namespace, "error", err)
-							// fast exit on permission denied
-							return err
-						}
-						log.Info("Cannot list imagestreams", "namespace", namespace, "error", err)
-						continue
+					if err != nil && k8serr.IsForbidden(err) {
+						log.Info("Permission denied to list imagestreams", "namespace", namespace, "error", err)
+						// fast exit on permission denied
+						return err
 					}
 
+					// List imagestreams in the specified notebook namespace
+					inNamespaceImagestreams, err := dynamicClient.Resource(ims).Namespace(notebook.namespace).List(ctx, metav1.ListOptions{})
+					if err != nil && k8serr.IsForbidden(err) {
+						log.Info("Permission denied to list imagestreams", "namespace", notebook.namespace, "error", err)
```

Comment:
We already have perms to list imagestreams, so no further additions to perms should be needed
https://github.com/opendatahub-io/kubeflow/blob/690694f9c7c4181e29915daed49e2f22b56fc913/components/odh-notebook-controller/config/rbac/role.yaml#L29-L35

Reply:
Ack, thanks 
changed it as per requirement